### DISCUSSION TITLE
Q: 怎么样减少waline 在vercel的部署占有内存？ || Q: How to reduce the memory occupied by waline in vercel deployment?

### DISCUSSION THREAD

**Discussion by appotry** [walinejs/waline#2062] [Q&A] [answered]
### Describe the bug | 问题描述

vercel 默认分配1G内存 ，但感觉waline 占有1G内存应该绰绰有余，应该可以分配更少的内存

小内存分配可以减少vercel 的 Functions-Execution 时间。 https://vercel.com/dashboard/usage#Serverless%20Functions-Execution

官方文档 如下 https://vercel.com/docs/projects/project-configuration#functions-property-with-serverless-functions

```
{
  "functions": {
    "api/test.js": {
      "memory": 3008,
      "maxDuration": 30
    },
    "api/*.js": {
      "memory": 3008,
      "maxDuration": 30
    }
  }
}
```

怎么样修改来减少waline 在vercel 的占有内存？

https://github.com/walinejs/waline/blob/main/example/index.js

### Website URL | 问题网站

waline.17lai.site

### Where your waline deploy? | 服务部署在哪里？

Vercel (Default)

### Where your comment data store? | 数据存储在哪里？

LeanCloud(https://leancloud.app)

---
### Describe the bug | Problem description

Vercel allocates 1G of memory by default, but I feel that 1G of memory for waline should be more than enough, and it should be possible to allocate less memory

Small memory allocation can reduce Vercel's Functions-Execution time. https://vercel.com/dashboard/usage#Serverless%20Functions-Execution

The official documentation is as follows https://vercel.com/docs/projects/project-configuration#functions-property-with-serverless-functions

```
{
  "functions": {
    "api/test.js": {
      "memory": 3008,
      "maxDuration": 30
    },
    "api/*.js": {
      "memory": 3008,
      "maxDuration": 30
    }
  }
}
```

How to modify to reduce the memory occupied by waline in vercel?

https://github.com/walinejs/waline/blob/main/example/index.js

### Website URL | Problem Website

waline.17lai.site

### Where your waline deploy? | Where is the service deployed?

Vercel (Default)

### Where your comment data store? | Where is your comment data store?

LeanCloud(https://leancloud.app)

**Comment by lizheming** [comment, marked as answer]
默认分配 1G 内存并不代表 Waline 使用了 1G 内存。你可以按照 Vercel 配置的形式限制执行文件的内存，不过内存占用和执行时间并没有严格的关系

---
The default allocation of 1G memory does not mean that Waline uses 1G memory. You can limit the memory of the executable file in the form of Vercel configuration, but there is no strict relationship between memory usage and execution time

**Reply by appotry** [reply]
但是vercel计算[Functions-Execution](https://vercel.com/dashboard/usage#Serverless%20Functions-Execution) 时间的时候是按照整体分配内存来计算的，并且每月限额免费100G HR时间。所以这里应该有优化空间

---
However, when vercel calculates [Functions-Execution](https://vercel.com/dashboard/usage#Serverless%20Functions-Execution) time, it is calculated according to the overall allocated memory, and the monthly limit is 100G HR time for free. So there should be room for optimization

**Comment by appotry** [comment]
https://vercel.com/guides/what-are-gb-hrs-for-serverless-function-execution

[Serverless Function Execution Example](https://vercel.com/guides/what-are-gb-hrs-for-serverless-function-execution#serverless-function-execution-example)
In this example, we will allocate 512 MB of memory and execute it 3 million times at a 1 second duration:

Total Seconds: 3M * (1s) = 3,000,000 Seconds

Total GB-Seconds: 512/1024 GB * 3,000,000 Seconds = 1,500,000 GB-Seconds

Total GB-Hrs: 1,500,000 GB-Seconds / 3600 = 416.67 GB-Hrs


The total Serverless Function Execution is 416.67 GB-Hrs.
